### PR TITLE
[HINTS] Add implementation for hint on assert_250_bit (math.cairo)

### PR DESCRIPTION
**Pull request by fmoletta**
# TITLE

## Description

Description of the pull request changes and motivation.

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: src/vm/hints/hint_utils.rs [hunk 1]
```diff
@@ -275,3 +278,78 @@ pub fn assert_le_felt(
...
+//        assert value < ids.UPPER_BOUND, f'{value} is outside of the range [0, 2**250).'
+//        # Calculation for the assertion.
+//        ids.high, ids.low = divmod(ids.value, ids.SHIFT)
+pub fn assert_250_bit(
+    vm: &mut VirtualMachine,
+    ids: HashMap<String, BigInt>,
+) -> Result<(), VirtualMachineError> {
+    //Declare constant values
+    let upper_bound = bigint!(2).pow(250);
+    let shift = bigint!(2).pow(128);
```
Comment (Oppen) [inline src/vm/hints/hint_utils.rs:294]: The hint takes these from the reference manager, are you sure it's OK to make it local constants?

Comment (fmoletta) [reply src/vm/hints/hint_utils.rs:294]: These values are not present in the reference ids, nor the reference_manager,
For example, from a compiled program using this hint, the reference_ids field is:
`"reference_ids": {
                        "starkware.cairo.common.math.assert_250_bit.high": 3,
                        "starkware.cairo.common.math.assert_250_bit.low": 2,
                        "starkware.cairo.common.math.assert_250_bit.range_check_ptr": 1,
                        "starkware.cairo.common.math.assert_250_bit.value": 0`
The values for SHIFT and UPPER_BOUND are stored in the identifiers section as constants:
`"starkware.cairo.common.math.assert_250_bit.SHIFT": {
            "type": "const",
            "value": 340282366920938463463374607431768211456
        },
        "starkware.cairo.common.math.assert_250_bit.SIZEOF_LOCALS": {
            "type": "const",
            "value": 0
        },
        "starkware.cairo.common.math.assert_250_bit.UPPER_BOUND": {
            "type": "const",
            "value": 1809251394333065553493296640760748560207343510400633813116524750123642650624
        },`
         and this identifiers section is not being fully parsed actually (only the pc of __main__ is saved and used).
         Moreover, this hint is very specific and is only used in the function assert_250_bit, which declares these same values as constants within the scope of the function. So while this solution  is  a bit messy , it shouldn't affect the behaviors of the math library

File: src/math_utils.rs [hunk 2]
```diff
@@ -3,6 +3,15 @@ use num_bigint::BigInt;
 use num_integer::Integer;
 use num_traits::{abs, FromPrimitive};
 
+/// Returns the lift of the given field element, val, as an integer in the range (-prime/2, prime/2).
+pub fn as_int(val: &BigInt, prime: &BigInt) -> BigInt {
+    if val < &prime.div_floor(&bigint!(2)) {
```
Comment (Oppen) [inline src/math_utils.rs:8]:
```suggestion
    if val < &prime.shr(1) {
```

File: src/vm/hints/hint_utils.rs (hunk 1 above)
Comment (Oppen) [inline src/vm/hints/hint_utils.rs:294]:
```suggestion
    let upper_bound = bigint!(1).shl(250);
    let shift = bigint!(1).shl(128);
```

File: src/math_utils.rs (hunk 2 above)
Comment (fmoletta) [reply src/math_utils.rs]: Fixed with merge

File: src/vm/hints/execute_hint.rs [hunk 3]
```diff
@@ -3290,4 +3293,139 @@ mod tests {
...
+            vm.segments.add(&mut vm.memory, None);
+        }
+        //Initialize fp
+        vm.run_context.fp = MaybeRelocatable::from((0, 3));
+        //Insert ids into memory
+        //ids.value
+        vm.memory
+            .insert(
+                &MaybeRelocatable::from((0, 0)),
+                &MaybeRelocatable::from(bigint!(2).pow(251)),
```
Comment (Oppen) [inline src/vm/hints/execute_hint.rs:3385]:
```suggestion
                &MaybeRelocatable::from(bigint!(1).shl(251)),
```

File: src/vm/hints/execute_hint.rs [hunk 4]
```diff
@@ -3290,4 +3293,139 @@ mod tests {
...
+                    offset2: 0,
+                    inner_dereference: false,
+                },
+            ),
+        ]);
+        //Execute the hint
+        assert_eq!(
+            execute_hint(&mut vm, hint_code, ids),
+            Err(VirtualMachineError::ValueOutside250BitRange(
+                bigint!(2).pow(251)
```
Comment (Oppen) [inline src/vm/hints/execute_hint.rs:3427]:
```suggestion
                bigint!(1).shl(251)
```

Comment (fmoletta) [reply src/vm/hints/execute_hint.rs:3427]: Is optimizing tests necessary?, Included change

**Review by Oppen — APPROVED** [review]

**Pull request merged**